assets/js/modules/search-console/util/report-date-range-args.test.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Tests for reporting date range utilities.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { generateDateRangeArgs } from './report-date-range-args';
+
+describe( 'Search Console reporting date range arguments', () => {
+	describe( 'generateDateRangeArgs', () => {
+		it( 'should throw if `startDate` or `endDate` arguments are not provided', () => {
+			try {
+				generateDateRangeArgs( { startDate: '2020-12-18' } );
+			} catch ( error ) {
+				expect( error.message ).toEqual( 'A valid endDate is required.' );
+			}
```

Comment:
This works but Jest provides a [`toThrow`](https://jestjs.io/docs/en/expect#tothrowerror) assertion which makes this a bit cleaner:

```suggestion
			expect(
				() => generateDateRangeArgs( { startDate: '2020-12-18' } )
			).toThrow( 'A valid endDate is required' )
```

Note the value for matching the exception message is optional and supports various methods.
So, not that we can't/shouldn't use try/catch for this, but we should use `toMatch` where possible (this is where we use elsewhere as well).